Source/WebKit/GPUProcess/graphics/WebGPU/RemoteGPU.cpp

```diff
@@ -75,10 +75,15 @@ void RemoteGPU::stopListeningForIPC(Ref<RemoteGPU>&& refFromConnection)
 {
     assertIsMainRunLoop();
     m_streamConnection->invalidate();
-    m_streamConnection->stopReceivingMessages(Messages::RemoteGPU::messageReceiverName(), m_identifier.toUInt64());
-    workQueue().dispatch([protectedThis = WTFMove(refFromConnection)]() {
-        protectedThis->workQueueUninitialize();
+
+    // This item is dispatched to the WorkQueue such that it will process it last, after any existing work.
+    m_workQueue->stopAndWaitForCompletion([&] {
+        workQueueUninitialize();
     });
+
+    m_streamConnection->invalidate();
```

Comment:
In general, this is incorrect way to do it, but it's hard to know since I haven't fixed the stuff. :( I have had #8002 pending but I got no reviewers on that one.